src/zope/app/apidoc/static.py

```diff
@@ -256,316 +314,297 @@ def sendMessage(self, msg, verbosity=4):
+        default=True,
+        help="""Output progress status."""
+    )
+
+    reporting.add_argument(
+        '--debug', '-d', action="store_true", dest='debug',
+        help="""Run in debug mode. This will allow you to use the debugger, if the publisher
+        experienced an error."""
+    )
+
+    return parser
 
 ######################################################################
 # Command-line processing
 
-# Default setup
-default_setup_args = [
-    '--verbosity', '5',
-    '--publisher',
-    '--url', 'http://localhost:8080/',
-    '--startpage', '++apidoc++/static.html',
-    '--username', 'mgr',
-    '--password', 'mgrpw',
-    '--progress',
-    '--add', '@@/varrow.png',
-    '--add', '@@/harrow.png',
-    '--add', '@@/tree_images/minus.png',
-    '--add', '@@/tree_images/plus.png',
-    '--add', '@@/tree_images/minus_vline.png',
-    '--add', '@@/tree_images/plus_vline.png',
-    '--ignore', 'twisted',
-    '--ignore', 'zope.app.twisted.ftp.test',
-    '--load-all'
-    ]
-
-def merge_options(options, defaults):
-    odict = options.__dict__
-    for name, value in defaults.__dict__.items():
-        if (value is not None) and (odict[name] is None):
-            odict[name] = value
-
-def get_options(args=None, defaults=None):
-
-    default_setup, _ = parser.parse_args(default_setup_args)
-    assert not _
-    if defaults:
-        defaults, _ = parser.parse_args(defaults)
-        assert not _
-        merge_options(defaults, default_setup)
-    else:
-        defaults = default_setup
 
-    if args is None:
-        args = sys.argv
-    original_testrunner_args = args
-    args = args[1:]
-    options, positional = parser.parse_args(args)
-    merge_options(options, defaults)
-    options.original_testrunner_args = original_testrunner_args
+def get_options(args=None):
+    #original_testrunner_args = args
 
-    if not positional:
-        parser.error("No target directory specified.")
-    options.target_dir = positional.pop()
+    options = _create_arg_parser().parse_args(args)
+
+    #options.original_testrunner_args = original_testrunner_args
 
     return options
 
 # Command-line UI
 ###############################################################################
 
 
-def main():
-    options = get_options()
-    maker = StaticAPIDocGenerator(options)
-    maker.start()
-    sys.exit(0)
+def main(args=None, generator=StaticAPIDocGenerator):
+    options = get_options(args)
+    maker = generator(options)
+    try:
+        # Replace a few things to make this work better.
+        # First, some scripts have names like __main__ and want to
+        # peek at sys.argv; arguments for us will not be correct
+        # for them, so we replace argv. Likewise, they may want to
+        # exit, and we don't want them to do that.
+        old_argv = sys.argv
+        sys.argv = ['program', '--help']
+
+        old_exit = sys.exit
+        def exit(_arg):
+            pass
+        sys.exit = exit
```

Comment:
Strongly suggest `try: ... except SystemExit: pass` instead.  Replacing sys.exit with a no-op would break anything that does

```python
...
if error_condition:
    sys.exit("cannot continue because blaaargh")
blithely_do_more_stuff()
...
```